compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -871,7 +871,41 @@ OMR::Power::TreeEvaluator::mAnyTrueEvaluator(TR::Node *node, TR::CodeGenerator *
 TR::Register*
 OMR::Power::TreeEvaluator::mAllTrueEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   return vloadEvaluator(node, cg);
+   TR::Node *inputNode = node->getFirstChild();
+   TR::Node *maskNode = node->getOpCode().isVectorMasked() ? node->getSecondChild() : NULL;
+
+   TR::Register *inputReg = cg->evaluate(inputNode);
+   TR::Register *maskReg = maskNode ? cg->evaluate(maskNode) : NULL;
+
+   TR::Register *resultReg = cg->allocateRegister(TR_GPR);
+   TR::Register *temp1 = cg->allocateRegister(TR_VRF);
+   TR::Register *temp2 = cg->allocateRegister(TR_VRF);
+
+   node->setRegister(resultReg);
+
+   if (maskReg)
+      generateTrg1Src3Instruction(cg, TR::InstOpCode::xxsel, node, temp1, inputReg, inputReg, maskReg);
+
+   //check if each half of input vector is all true values separately
+   generateTrg1ImmInstruction(cg, TR::InstOpCode::vspltisw, node, temp2, -1);
+   generateTrg1Src2Instruction(cg, OMR::InstOpCode::vcmpequd, node, temp1, temp1, temp2);
+
+   //AND the two halves together (i.e.: both halves must contain all true values for AllTrue to return true)
+   generateTrg1Src2ImmInstruction(cg, OMR::InstOpCode::xxpermdi, node, temp2, temp1, temp1, 2);
+   generateTrg1Src2Instruction(cg, OMR::InstOpCode::vand, node, temp1, temp1, temp2);
+
+   //move result to GPR
+   generateTrg1Src1Instruction(cg, TR::InstOpCode::mfvsrd, node, resultReg, temp1);
+
+   //AND with 0000...01 to ensure we return 1 if true, 0 if false
+   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::andi_r, node, resultReg, resultReg, 1);
+
```

Comment:
you can do better.  `vpopcntw` followed by `vsumsws`, then moving back to GPR, then shift right 7 bits to get your result.  for all 1 mask, the popcnt is 128.